fastlane/lib/fastlane/actions/ensure_git_status_clean.rb

```diff
@@ -74,8 +75,14 @@ def self.available_options
                                        type: Boolean),
           FastlaneCore::ConfigItem.new(key: :ignored,
                                        env_name: "FL_ENSURE_GIT_STATUS_CLEAN_IGNORED_FILE",
-                                       description: "The flag whether to ignore file the git status if the repo is dirty",
-                                       optional: true)
+                                       description: "The ignored files handling mode if the repo is dirty. The available options are: 'traditional', 'none' (default) and 'matching'",
```

Comment:
It's not clear what each of them does 🤔 

Since we're just forwarding a string down to another API (git), we could either copy the documentation from git itself, or simply point users to it. What do you think? I feel like this new docs are clearer than the previous one for sure, but still not clear enough, and perhaps even wrong? [git docs](https://git-scm.com/docs/git-status#Documentation/git-status.txt---ignoredltmodegt) shows that the this option is optional and that it defaults to `traditional` rather than `no` 🤔 

This change is also a breaking one, because there may be users that used to pass `no` here and now `no` isn't a valid option anymore 😬 

I particularly find git's docs more clear

Reply:
Hi and thanks for your review. I would answer to with yes and no. 

The fact of the matter is that before the change from `repo_status = Actions.sh("git status --porcelain --ignored #{ignored_mode}")` to `repo_status = Actions.sh("git status --porcelain --ignored='#{ignored_mode}'")` it didn't matter what argument you passed to `ensure_git_status_clean ignored:''` because the argument was ignored since the `git status` call was just wrong . 
This comment (https://github.com/fastlane/fastlane/issues/19818#issuecomment-1032355009) suggest fastlane is parsing unknown or empty arguments to 'no' instead of the git default 'traditional'. And I tested this (see above), not passing `--ignored=' '` to `git status --porcelain` does not give you the same results as passing `--ignored='tradition'`. It seems to be `--ignored='no'`. Hence the result of using `ensure_git_status_clean` will always give you the same result as calling `git status --porcelain` aka. `git status --porcelain --ignored='no'`.

Reply:
> or simply point users to it

I did this, so we have a single source of truth.

---

> and that it defaults to `traditional` rather than `no`

That's what I thought too. But if you read it carefully, you'll see that it's not like that:
- It says that there is an `--ignored` argument, with an optional `mode` parameter (because it's in square brackets):
  > `--ignored[=<mode>]`
- Then, it says what the `mode` parameter (not the `--ignored` argument altogether) does:
  > The mode parameter is used to specify the handling of ignored files.
- And then, when it says that:
  > It is optional

  it still refers to the `mode` parameter, as well as the continuation:
  > it defaults to `traditional`.

So, when the `mode` parameter is not specified, because it's optional, it defaults to `traditional`, meaning, it would be the same as passing it explicitly to the `--ignored` argument like this: `--ignored='traditional'`.

But not specifying the `--ignored` argument at all, is sort of like using a `--no-ignored` argument (which doesn't exist), so it just makes sense for this to behave like `--ignored='no'`.

_(Actually, it would have made more sense for the `no` option to not exist at all, but 🤷🏻‍♂️)._

The `--ignored` argument has the description:
> Show ignored files as well.

and the `no` mode parameter has the description:
> Show no ignored files.

so this, too, shows that this parameter cancels the argument altogether.

---

> This change is also a breaking one, because there may be users that used to pass no here and now no isn't a valid option anymore

True, but it didn't work at all before this 🤔 Do you know if we mark breaking changes in some way?

---

> I particularly find git's docs more clear

I tried to make as few changes as possible to the existing docs. I wrote them from scratch now 😁 Let me know if they're ok or not.

Reply:
> and that it defaults to `traditional` rather than `no`

That's what I thought too. But if you read it carefully, you'll see that it's not like that:
- It says that there is an `--ignored` argument, with an optional `mode` parameter (because it's in square brackets):
  > `--ignored[=<mode>]`
- Then, it says what the `mode` parameter (not the `--ignored` argument altogether) does:
  > The mode parameter is used to specify the handling of ignored files.
- And then, when it says that:
  > It is optional

  it still refers to the `mode` parameter, as well as the continuation:
  > it defaults to `traditional`.

So, when the `mode` parameter is not specified, because it's optional, it defaults to `traditional`, meaning, it would be the same as passing it explicitly to the `--ignored` argument like this: `--ignored='traditional'`.

But not specifying the `--ignored` argument at all, is sort of like using a `--no-ignored` argument (which doesn't exist), so it just makes sense for this to behave like `--ignored='no'`.

_(Actually, it would have made more sense for the `no` option to not exist at all, but 🤷🏻‍♂️)._

The `--ignored` argument has the description:
> Show ignored files as well.

and the `no` mode parameter has the description:
> Show no ignored files.

so this, too, shows that this parameter cancels the argument altogether.

Reply:
> This change is also a breaking one, because there may be users that used to pass no here and now no isn't a valid option anymore

True, but it didn't work at all before this 🤔 Do you know if we mark breaking changes in some way?

Reply:
> I particularly find git's docs more clear

I tried to make as few changes as possible to the existing docs. I wrote them from scratch now 😁 Let me know if they're ok or not.

Reply:
Thanks for the thorough responses 😁 

- Default value: that's mindblowing 😂 your rationale makes sense now. Git docs were poorly written 😭 
- Breaking changes: if the previous implementation didn't work at all, I believe there's no harm in making this change, even if it's a breaking one. After all, you're making it actually do something and work now so, I don't see harm in it.
- Your new docs: LGTM! 😊